### ISSUE TITLE
Add support for Azardi

### ISSUE THREAD

**Issue by nvdaes** [nvaccess/nvda#5848]
Azardi is anebook reader which supports ePUB format, including media overlays.
XHTML documents are shown in a Mozilla window, but browse mode doesn't work in NVDA.
Also, it's needed to press NVDa+F2 to access some features of the application, as explained in the Accessibility tips section of the DAISY Consortium website at:
http://www.daisy.org/daisypedia/azardi-epub-reader-windows
Thanks.

**Comment by nvdaes** [comment]
Excuse me, after testing this program in more detail, I think that pressing NVDA+f2 can be avoid at this moment.
This is used to navigate to the previous and next chapter ob books, but it can be done just pressing NVDA plus the corresponding keystroke of Azardi, that is, NVDA+shift + page up and down.
I think that the only needed thing is the support for browse mode.
For usability, imo it's recommended to add Enter and applications keys to the tree which shows the list of added books to the library, for opening the book and the context menu.
I have created a global plugin since I can't use an ap module, perhaps since the app name is azardi-2.0 and I don't know how to work arround this without a global plugin.
It has an issue when using the searching feature of Azardi: if I move to the results and then I perform mouse clicks to move to the corresponding chapter, when I open a dialog or a menu over the application and then I close this NVDA gui, the caret seems to disappear.
It can be tested pressing NVDA+control+f to search text. The string can be found, but then the caret can't be moved since the focus is placed again in the document, for instance with alt+tab.
Perhaps it's due to the custom class assignment, at least in the plugin.
Furthermore, here I don't have success using findDescendantWindow to move to the toolbar and so on, though I think this is not so important.
I will attach the plugin in a zip file, in case it's useful.
Thanks.
[azardi2.zip](https://github.com/nvaccess/nvda/files/201502/azardi2.zip)
[azardi2.zip](https://github.com/nvaccess/nvda/files/201503/azardi2.zip)

**Comment by jcsteh** [comment]
I'm happy to accept all of these changes, but we definitely need to get 
this into an app module, not a global plugin. The trouble is that the 
dot (".") conflicts with Python's use of "." in package paths. I guess 
we'll need to convert this to an underscore or similar in appModuleHandler.

**Comment by nvdaes** [comment]
Hi, I'm committed changes for this on a branch:
1. I have modified appModuleHandler replacing dot with underscore to 
test azardi_2.0 module.
This is requested on #5323, which imo should block this one.
2. I have added azardi_2.0 app module with copiright headers (feel free 
to add NVDA contributors or whatever you want, though I have added my 
full name thinking that you may require it later).
3. I have added key commands for Azardi in the applications section of 
the User guide.
Here I don't know if we should mention that NVDA+shift+page up and down 
can be used when in a document to navigate throught book chapters:
It's not an added script, but it works by default.
Enter and application keys can be used, thought can't be changed. I 
think that here it doesn't make too much sense to add script 
documentation and categories, as they are intuitive keystrokes.
When all this is clarified, I can continue and make a pull request.
Thanks.

**Comment by nvdaes** [comment]
The mentioned branch is at
https://github.com/nvdaes/nvda/tree/i5848

**Comment by nvdaes** [comment]
I have commited a new change, choosing the AzardiDocument class to add
role_document when role is set to frame.
In this way the toolbar and Azardi help (which is shown when pressing
f1) are accessible, and the search feature can be used too.
I think that we shouldn't add more information to the user guide.
So, if you don't have objections, I will send a pull request later.
Thanks.

**Comment by nvdaes** [comment]
A pull request for this is created at
https://github.com/nvaccess/nvda/pull/5858
You can review when you want and feel free to change whatever it's
necessary.
Thanks.

**Comment by bhavyashah** [comment]
A friendly ping to @jcsteh, @feerrenrut and other reviewers to take a look at the PR submitted by @nvdaes , since it seems to be almost four months old.

**Comment by nvdaes** [comment]
Hi, I think it's better to look at the issue related to VitalSource,
better evaluated than Azardi regarding to accessibility for reading
ePUB. I use vitalSource now and not Azardi, and finally I checked that
the problem regarding to focus doesn't occur with Narrator.
Cheers.

2017-08-02 9:10 GMT+02:00, bhavyashah <notifications@github.com>:
